Emby.Server.Implementations/Updates/InstallationManager.cs

```diff
@@ -84,230 +85,124 @@ public class InstallationManager : IInstallationManager
-                {
-                    continue;
-                }
-
-                // If an app version was supplied, filter the versions for each package to only include supported versions
-                if (applicationVersion != null)
-                {
-                    p.versions = p.versions.Where(v => IsPackageVersionUpToDate(v, applicationVersion)).ToArray();
-                }
-
-                if (p.versions.Length == 0)
-                {
-                    continue;
-                }
-
-                returnList.Add(p);
+                var strGuid = guid.ToString("N", CultureInfo.InvariantCulture);
+                availablePackages = availablePackages.Where(x => x.guid.Equals(strGuid, StringComparison.OrdinalIgnoreCase));
             }
 
-            return returnList;
+            return availablePackages;
         }
 
-        /// <summary>
-        /// Determines whether [is package version up to date] [the specified package version info].
-        /// </summary>
-        /// <param name="packageVersionInfo">The package version info.</param>
-        /// <param name="currentServerVersion">The current server version.</param>
-        /// <returns><c>true</c> if [is package version up to date] [the specified package version info]; otherwise, <c>false</c>.</returns>
-        private static bool IsPackageVersionUpToDate(PackageVersionInfo packageVersionInfo, Version currentServerVersion)
+        /// <inheritdoc />
+        public IEnumerable<PackageVersionInfo> GetCompatibleVersions(
+            IEnumerable<PackageVersionInfo> availableVersions,
+            Version minVersion = null,
+            PackageVersionClass classification = PackageVersionClass.Release)
         {
-            if (string.IsNullOrEmpty(packageVersionInfo.requiredVersionStr))
+            var appVer = _applicationHost.ApplicationVersion;
+            availableVersions = availableVersions
+                .Where(x => x.classification == classification
+                    && Version.Parse(x.requiredVersionStr) <= appVer);
+
+            if (minVersion != null)
             {
-                return true;
+                availableVersions = availableVersions.Where(x => x.Version >= minVersion);
             }
 
-            return Version.TryParse(packageVersionInfo.requiredVersionStr, out var requiredVersion) && currentServerVersion >= requiredVersion;
+            return availableVersions.OrderByDescending(x => x.Version);
         }
 
-        /// <summary>
-        /// Gets the package.
-        /// </summary>
-        /// <param name="name">The name.</param>
-        /// <param name="guid">The assembly guid</param>
-        /// <param name="classification">The classification.</param>
-        /// <param name="version">The version.</param>
-        /// <returns>Task{PackageVersionInfo}.</returns>
-        public async Task<PackageVersionInfo> GetPackage(string name, string guid, PackageVersionClass classification, Version version)
```

Comment:
Was this method unused before you removed it?

Reply:
No, but it's got replaced by `GetCompatibleVersions`